src/package-resolver.js

```diff
@@ -496,18 +495,14 @@ export default class PackageResolver {
       return;
     }
 
-    const request = new PackageRequest(req, this);
-    const fetchKey = `${req.registry}:${req.pattern}:${String(req.optional)}`;
-    const initialFetch = !this.fetchingPatterns.has(fetchKey);
+    const existingManifest = this.patterns[req.pattern];
     let fresh = false;
 
     if (this.activity) {
       this.activity.tick(req.pattern);
     }
-
-    if (initialFetch) {
-      this.fetchingPatterns.add(fetchKey);
```

Comment:
If you remove this pqrt, there will be multiple concurrent request for the same pattern, is that correct? Since `this.patterns[req.pattern]` won't be populated until the request resolves.